jsparagus/emit/rust.py

```diff
@@ -102,7 +102,7 @@ def write(self, indentation, string, *format_args):
     def header(self):
         self.write(0, "// WARNING: This file is autogenerated.")
         self.write(0, "")
-        self.write(0, "use ast::{arena::{Box, Vec}, types::*};")
```

Comment:
Let's avoid renaming this API. In `ast/lib.rs`, we can have

```
#[path = "types_generated.rs"]
mod types;
```

Or, if you don't like that, equivalently

```
mod types_generated;
pub mod types {
    pub use types_generated::*;
}
```

Reply:
Generally, consider this approach for all public modules (used directly by other crates). Crates' public features should not contain `_generated` as that is an implementation detail.

Remember, we want to drop ast.json and use ast/types.rs as the source of truth (with Rust `derive` macros to generate any other code we need), so we would just change this back.